ape_safe/client/__init__.py

```diff
@@ -186,6 +189,53 @@ def estimate_gas_cost(
         gas = result.get("safeTxGas")
         return int(HexBytes(gas).hex(), 16)
 
+    def get_delegates(self) -> Dict[AddressType, List[AddressType]]:
+        url = "delegates"
+        delegates: Dict[AddressType, List[AddressType]] = {}
+
+        while url:
+            response = self._get(url, params={"safe": self.address})
+            data = response.json()
+
+            for delegate_info in map(DelegateInfo.model_validate, data.get("results", [])):
+                if delegate_info.delegator not in delegates:
+                    delegates[delegate_info.delegator] = [delegate_info.delegate]
+                else:
+                    delegates[delegate_info.delegator].append(delegate_info.delegate)
+
+            url = data.get("next")
+
+        return delegates
+
+    def add_delegate(self, delegate: AddressType, label: str, delegator: AccountAPI):
+        msg_hash = self.create_delegate_message(delegate)
+
+        # NOTE: This is required as Safe API uses an antiquated .signHash method
+        if not (sig := delegator.sign_raw_msghash(msg_hash)):
+            raise ActionNotPerformedError("Did not sign delegate approval")
+
+        payload = {
+            "safe": self.address,
+            "delegate": delegate,
+            "delegator": delegator.address,
+            "label": label,
+            "signature": sig.encode_rsv().hex(),
+        }
+        self._post("delegates", json=payload)
+
+    def remove_delegate(self, delegate: AddressType, delegator: AccountAPI):
+        msg_hash = self.create_delegate_message(delegate)
+
+        # NOTE: This is required as Safe API uses an antiquated .signHash method
+        if not (sig := delegator.sign_raw_msghash(msg_hash)):
+            raise ActionNotPerformedError("Did not sign delegate removal")
+
+        payload = {
+            "delegator": delegator.address,
+            "signature": sig.encode_rsv().hex(),
+        }
+        self._delete(f"delegates/{delegate}", json=payload)
```

Comment:
**warning**: trailing slashes are important i think for all of Safe's APIs

Reply:
true, the api states it needs a trailing slash https://safe-transaction-mainnet.safe.global/